onnxruntime/core/graph/model.cc

```diff
@@ -233,10 +236,13 @@ Model::Model(ModelProto&& model_proto, const PathString& model_path,
     model_local_function_templates_maps_.insert_or_assign(function_utils::GetFunctionIdentifier(func.domain(), func.name()), std::move(func_template_ptr));
   }
 
+  std::function<void()> model_functions_remover = [this]() {
+    RemoveLocalFunctionsProtos();
+  };
   // create instance. need to call private ctor so can't use make_unique
   GSL_SUPPRESS(r.11)
   graph_.reset(new Graph(*this, model_proto_.mutable_graph(), domain_to_version, IrVersion(), schema_registry,
-                         logger, options.strict_shape_type_inference));
+                         logger, options.strict_shape_type_inference, std::move(model_functions_remover)));
```

Comment:
minor question: can't we just pass `this->RemoveLocalFunctionsProtos` directly (without having to explicitly construct the closure `model_functions_remover` above)? Anyway: the more important question is the one below, whether we want to remove all functions or only those that were inlined.

Reply:
This we cannot do, since the model is exposed as const.